BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java

```diff
@@ -0,0 +1,144 @@
+package de.bibbuddy;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.util.SparseArray;
+import android.view.LayoutInflater;
+import android.view.SurfaceHolder;
+import android.view.SurfaceView;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.app.ActivityCompat;
+import androidx.fragment.app.Fragment;
+import com.google.android.gms.vision.CameraSource;
+import com.google.android.gms.vision.Detector;
+import com.google.android.gms.vision.barcode.Barcode;
+import com.google.android.gms.vision.barcode.BarcodeDetector;
+import java.io.IOException;
+
+/**
+ * The BookBarcodeScannerFragment is responsible to scan the ISBN of the book
+ * that a user wants to add to a shelf.
+ *
+ * @author Claudia Schönherr
+ */
+public class BookBarcodeScannerFragment extends Fragment {
+  private static final int REQUEST_CAMERA_PERMISSION = 201;
+
+  private SurfaceView surfaceView;
+  private CameraSource cameraSource;
+  private TextView barcodeText;
+  private String barcodeData;
+  private BarcodeDetector barcodeDetector;
+
+  private Long shelfId;
+  private String shelfName;
+
+  @Nullable
+  @Override
+  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                           @Nullable Bundle savedInstanceState) {
+    View view = inflater.inflate(R.layout.fragment_barcode_scanner, container, false);
+
+    surfaceView = view.findViewById(R.id.surface_view);
+    barcodeText = view.findViewById(R.id.barcode_value);
+    Bundle bundle = getArguments();
+
+    shelfId = bundle.getLong(LibraryKeys.SHELF_ID);
+    shelfName = bundle.getString(LibraryKeys.SHELF_NAME);
+
+    setupDetectorsAndSources(view);
+    ((MainActivity) getActivity()).updateHeaderFragment(getString(R.string.isbn_scan));
+
+    return view;
+  }
+
+  private void setupDetectorsAndSources(View view) {
+    barcodeDetector = new BarcodeDetector.Builder(view.getContext())
+        .setBarcodeFormats(Barcode.ISBN | Barcode.EAN_13) // TODO more barcode formats?
```

Comment:
I dont think more are needed at the moment - ISBNs are standardized so that should do the job :)